plugins/ImageGraph/API.php

```diff
@@ -129,404 +130,453 @@ public function get(
         $segment = false
     )
     {
-        Piwik::checkUserHasViewAccess($idSite);
-
-        // Health check - should we also test for GD2 only?
-        if (!SettingsServer::isGdExtensionEnabled()) {
-            throw new Exception('Error: To create graphs in Piwik, please enable GD php extension (with Freetype support) in php.ini,
-            and restart your web server.');
-        }
-
-        $useUnicodeFont = array(
-            'am', 'ar', 'el', 'fa', 'fi', 'he', 'ja', 'ka', 'ko', 'te', 'th', 'zh-cn', 'zh-tw',
-        );
-        $languageLoaded = Translate::getLanguageLoaded();
-        $font = self::getFontPath(self::DEFAULT_FONT);
-        if (in_array($languageLoaded, $useUnicodeFont)) {
-            $unicodeFontPath = self::getFontPath(self::UNICODE_FONT);
-            $font = file_exists($unicodeFontPath) ? $unicodeFontPath : $font;
-        }
-
-        // save original GET to reset after processing. Important for API-in-API-call
-        $savedGET = $_GET;
-
-        try {
-            $apiParameters = array();
-            if (!empty($idGoal)) {
-                $apiParameters = array('idGoal' => $idGoal);
+        $idSiteArr = array();
+        if (strpos($idSite, ',') !== false) {
+            $idSiteList = explode(',', $idSite);
+            foreach ($idSiteList as $key) {
+                $idSiteArr[] = $key;
             }
-            // Fetch the metadata for given api-action
-            $parameters = array(
-                'idSite' => $idSite,
-                'apiModule' => $apiModule,
-                'apiAction' => $apiAction,
-                'apiParameters' => $apiParameters,
-                'language' => $languageLoaded,
-                'period' => $period,
-                'date' => $date,
-                'hideMetricsDoc' => false,
-                'showSubtableReports' => true
-            );
-
-            $metadata = Request::processRequest('API.getMetadata', $parameters);
-            if (!$metadata) {
-                throw new Exception('Invalid API Module and/or API Action');
+        } else if (!empty($idSite) && ($idSite != 'all')) {
+            $idSiteArr[] = $idSite;
+        } else if (!empty($idSite) && ($idSite == 'all')) {
+            $viewableIdSites = APISitesManager::getInstance()->getSitesIdWithAtLeastViewAccess();
+            foreach ($viewableIdSites as $site) {
+                $idSiteArr[] = $site;
             }
+        } else {
+            // empty or invalid idSite has been already handled somewhere
+            exit;
```

Comment:
Yes, I would remove exit else loop from here. It already throws exception before it reaches this point.